### ISSUE TITLE
generate signature for SparseTensor failed

### ISSUE THREAD

**Issue by ericyue** [tensorflow/tensorflow#6110]
### What related GitHub issues or StackOverflow threads have you found by searching the web for your problem?
I can not find any posts about how to export a sparse model in tensorflow. I searched for hours.

### Environment info
Operating System:
centos7 without gpu

### If possible, provide a minimal reproducible example (We usually don't have time to read hundreds of lines of your code)
export function:
```
# Read TFRecords files for training
filename_queue = tf.train.string_input_producer(
    tf.train.match_filenames_once(FLAGS.train),
    num_epochs=epoch_number)
serialized_example = read_and_decode(filename_queue)
batch_serialized_example = tf.train.shuffle_batch(
    [serialized_example],
    batch_size=batch_size,
    num_threads=thread_number,
    capacity=capacity,
    min_after_dequeue=min_after_dequeue)
features = tf.parse_example(
    batch_serialized_example,
    features={
        "label": tf.FixedLenFeature([], tf.float32),
        "ids": tf.VarLenFeature(tf.int64),
        "values": tf.VarLenFeature(tf.float32),
    })
batch_labels = features["label"]
batch_ids = features["ids"]
batch_values = features["values"]

        print("Start to run export model")
        start_time = datetime.datetime.now()

        ckpt = tf.train.get_checkpoint_state(checkpoint_dir)
        if ckpt and ckpt.model_checkpoint_path:
            print("Use the model {}".format(ckpt.model_checkpoint_path))
            saver.restore(sess, ckpt.model_checkpoint_path)
            export_version = FLAGS.export_version
            export_path = "./export/"+job_name
            print('Exporting trained model to %s' % export_path)
            model_exporter = exporter.Exporter(saver)
            model_exporter.init(
                sess.graph.as_graph_def(),
                named_graph_signatures = {
                  'inputs': exporter.generic_signature({'ids': features['ids'], 'values':features['values']}),
                  'outputs': exporter.generic_signature({'labels': features['labels']})},
                init_op=init_op)
            model_exporter.export(export_path, tf.constant(export_version), sess)
            print('Done exporting!')
```

### Logs or other output that would be helpful
(If logs are large, please upload as attachment or provide link).
```
Traceback (most recent call last):
  File "run.py", line 343, in <module>
    'inputs': exporter.generic_signature({'ids': features['ids'], 'values':features['values']}),
  File "/home/serving/anaconda2/lib/python2.7/site-packages/tensorflow/contrib/session_bundle/exporter.py", line 121, in generic_signature
    signature.generic_signature.map[name].tensor_name = tensor.name
AttributeError: 'SparseTensor' object has no attribute 'name'
```

**Comment by nubbel** [comment]
@ericyue Could you find a solution?

**Comment by jschenxiaoyu** [comment]
Also meet this problem. Any suggestion? @ericyue

**Comment by Qululu** [comment]
I'm getting the same error when calling `tf.python.saved_model.utils.build_tensor_info()` on a `sparse_placeholder` (TF v1.0). Anyone have a solution yet?

**Comment by shahinkl** [comment]
can we use like this?
`inputs = {'source': utils.build_tensor_info(pred_data[0]),
              'target_sparse': (utils.build_tensor_info(pred_data[1].indices), utils.build_tensor_info(pred_data[1].values), utils.build_tensor_info(pred_data[1].dense_shape)),
              'source_len': utils.build_tensor_info(pred_data[2])
              }`

**Comment by kernelmode** [comment]
Why this issue was closed? I am experiencing the same problem.

**Comment by dpressel** [comment]
I think the details of the original issue are out of date, but it still doesnt seem possible to use sparse tensors as part of a signature.  Is this intended behavior?

From this doc:

https://github.com/tensorflow/serving/blob/master/tensorflow_serving/g3doc/signature_defs.md

"Note that TensorInfo itself requires specification of name, dtype and tensor shape. While tensor information is already present in the graph, it is useful to explicitly have the TensorInfo defined as part of the SignatureDef since tools can then perform signature validation, etc. without having to read the graph definition."

(linked from here: https://github.com/tensorflow/tensorflow/tree/master/tensorflow/python/saved_model)

This seems consistent with the code -- looking here (in  `def _validate_signature_def_map(self, signature_def_map)`):

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/saved_model/builder_impl.py#L202

The definition of TensorInfo seems to show that sparse will not have a name:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/protobuf/meta_graph.proto

So basically, when using a VarLenFeature, export fails:
```
AssertionError: All TensorInfo protos used in the SignatureDefs must have the name field set: dtype: DT_STRING
```

This fix from a few days ago in `saved_model.utils.py` seems to imply that using a SparseTensor should be possible, but maybe I'm misinterpreting:

https://github.com/tensorflow/tensorflow/commit/eb27277809c038d93edcfb7221295608654589d7

**Comment by formath** [comment]
I also meet the similar problem. Solutions about prediction using sparse tensor are really really rare.

**Comment by XuesongYang** [comment]
same here, and still looking forward to the solutions.

**Comment by ksbg** [comment]
I am having the same problem. I can't find a way to export using SavedModelBuilder with signatures generated from SparseTensors. Is there really no way of achieving this?

**Comment by MatteoGlarey** [comment]
I have the same issue, I am working with sparse_placeholder and I get this error:
AssertionError: All TensorInfo protos used in the SignatureDefs must have the name field set: dtype: DT_FLOAT
tensor_shape {
  unknown_rank: true
}
coo_sparse {
  values_tensor_name: "input/values:0"
  indices_tensor_name: "input/indices:0"
  dense_shape_tensor_name: "input/shape:0"
}

**Comment by ksbg** [comment]
@MatteoGlarey I "solved" the problem by building tensor infos from the 3 individual Tensors that make up a SparseTensor (*/indices, */values, */shape) and then save the model using these tensor infos. The downside is that when the model is being deployed using Tensorflow Serving, the value to be scored has to be supplied in that same format, meaning as incides/values/shape individually.

**Comment by MatteoGlarey** [comment]
Thank you for the fast reply, I will try it as soon as possible!
